lib/SILGen/SILGenFunction.cpp

```diff
@@ -157,6 +158,57 @@ DeclName SILGenModule::getMagicFunctionName(SILDeclRef ref) {
   llvm_unreachable("Unhandled SILDeclRefKind in switch.");
 }
 
+void SILGenFunction::enterDebugScope(SILLocation Loc, bool isBindingScope,
+                                     Optional<SILLocation> MacroExpansion,
+                                     DeclNameRef MacroName,
+                                     DeclNameLoc MacroNameLoc) {
+  auto *Parent = DebugScopeStack.size() ? DebugScopeStack.back().getPointer()
+                                        : F.getDebugScope();
+  auto *Scope = Parent;
+  // Don't nest a scope for Loc under Parent unless it's actually different.
+  if (RegularLocation(Parent->getLoc()) != RegularLocation(Loc)) {
+    SILDebugScope *InlinedAt = nullptr;
+    // Create an inline scope for a macro expansion.
+    if (MacroExpansion && MacroName && MacroNameLoc.isValid()) {
+      InlinedAt = new (SGM.M) SILDebugScope(RegularLocation(*MacroExpansion),
+                                            &getFunction(), Parent);
+      SILGenFunctionBuilder B(SGM);
+      auto ExtInfo = SILFunctionType::ExtInfo::getThin();
+      auto FunctionType = SILFunctionType::get(
+          nullptr, ExtInfo, SILCoroutineKind::None,
+          ParameterConvention::Direct_Unowned, /*Params*/ {}, /*yields*/ {},
+          /*Results*/ {}, None, SubstitutionMap(), SubstitutionMap(),
+          SGM.M.getASTContext());
+      SILFunction *MacroFn = B.getOrCreateFunction(
+          Loc, MacroName.getBaseIdentifier().str(),
```

Comment:
Should we render the full macro name here, with argument labels? And is this something that will eventually need to be unique?

Reply:
Ideally, we should provide the fully mangled name here, then LLDB can render it appropriately. It would also be awesome to emit pattern_binding nodes for all the arguments (if we don't do that already) and associate them with macro arguments as (`debug_value(_addr) %0 : $Type, let, name "arg_1", argno 1, loc ...`) in SIL.